snapcraft/internal/meta.py

```diff
@@ -137,7 +139,17 @@ def _record_snapcraft(self):
         if os.environ.get('SNAPCRAFT_BUILD_INFO'):
             os.makedirs(record_dir, exist_ok=True)
             with open(record_file_path, 'w') as record_file:
-                yaml.dump(self._config_data, record_file)
+                annotated_snapcraft = self._annotate_snapcraft(
+                    copy.deepcopy(self._config_data))
+                yaml.dump(annotated_snapcraft, record_file)
+
+    def _annotate_snapcraft(self, data):
+        for part in data['parts']:
+            pull_state = get_state(
```

Comment:
makes a lot of sense, but that IMO is a different branch because it will touch many other files.

@sergiusens would you like it as a prerequisite of this one, or can I add it to my pile and do it after all the existing ones are merged?